accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
+Extension members can be invoked *implicitly* (without mentioning the extension by name) as if they were members of the `on` type of the extension. This is intended as the primary way to use extensions, with explicit extension member invocation as a fallback for cases where the implicit extension resolution doesn't do what the user want. 
 
-Example:
+An implicit extension member invocation occurs for a simple or composite member invocation with a target expression `e` iff there exists a unique *most specific* extension declaration which is *accessible* and *applicable* to the member invocation (see below).
 
-```dart
-extension SuperList<T> on List<T> {
-  R foldRight<R>(R base, R combine(T element, R accumulator)) {
-    for (int i = this.length - 1; i >= 0; i--) {
-      base = combine(this[i], base);
-    }
-    return base;
-  }
-}
-...
-  List<String> strings = ...;
-  int count(String string, int length) => length + string.length;
-  ...
-  var length = strings.foldRight(0, count);
-```
+If `E` is the single most specific accessible and applicable extension for a member invocation *i* with target expression `e`, which would then be a compile-time error by itself,  then we treat the target expression as if it was the extension application of `E` to `e`, and if `E` is generic, also providing the type arguments inferred for `E` in checking that it was applicable. This makes the member invocation equivalent to an explicit extension member invocation. This happens even if the *name* `E` is not accessible, so this is not a syntactic rewrite.
 
-Here the inference occurs just as if the extension had been declared like:
+Implicit extension member invocation can also apply to individual *cascade* invocations. A cascade is treated as if each cascade section was a separate member invocation on an expression with the same value as the cascade receiver expression (the one before the first `..`). This means that a cascade like `o..foo()..bar()` may perform an implicit extension member invocation on `o` for `foo()` and a normal invocation on `o` for `bar()`. There is no way to specify the corresponding explicit member invocation, so this is again not a syntactic rewrite.
 
-```dart
-class SuperList<T> {
-  final List<T> $this;
-  SuperList(this.$this);
-  R foldRight<R>(R base, R Function(T, R) combine) { ... }
-}
-```
+##### Accessibility
 
-and it was invoked as `SuperList(strings).foldRight(0, count)`.
+An extension is *accessible* for an expression if it has a name which is not private to a different library and the extension declaration is declared or imported into a scope which is a (recursive) parent scope of the lexical scope of the expression. That currently means imported into the default import scope of the current library, or declared at the top-level of the library itself.
```

Comment:
Even mentioning that the extension has a name which is private to another library breaks the rules for namespace management: Otherwise, we never need to even consider what roles such a name could play.

So the cost associated with this kind of language is that we must start from scratch, leaving all scope rules etc. in the dust.

A fresh name of the extension would allow us to rely on standard namespace management, which would also allow us to connect to other parts of the language, e.g., when an explicitly resolved extension method invocation is inferred and checked as-if via a corresponding class. Otherwise that class would also have a name that "doesn't work" according to the standard scope-and-namespace rules.

Reply:
Rewritten to not mention scopes.